fast_llm/models/gpt/model.py

```diff
@@ -63,6 +70,10 @@ def __init__(
         if self._config.enable_dpo:  # TODO better way to pass in?
             self._preprocessors.append(PreferenceSpanPreprocessor(self._config, self._tensor_space))
 
+        if self._config.vision_encoder.enabled:
```

Comment:
I think this would be more appropriate as a separate model, like we did for SSM

Reply:
No, this is intentionally part of the base GPT model. It ensures that multimodal support is transparent and inherited by all model variants (including SSM) without needing separate parallel class hierarchies. We're not introducing architectural silos unless there's a concrete need.